youtube_dl/extractor/abc.py

```diff
@@ -105,22 +105,53 @@ def _real_extract(self, url):
 
 class ABCIViewIE(InfoExtractor):
     IE_NAME = 'abc.net.au:iview'
-    _VALID_URL = r'https?://iview\.abc\.net\.au/(?:[^/]+/)*video/(?P<id>[^/?#]+)'
+    _VALID_URL = r'https?://iview\.abc\.net\.au/((?:[^/]+/)*video/|programs/(?:[^/]+/)*)(?P<id>[A-Z0-9]+)'
```

Comment:
1. Do not restrict `id` expression.
2. Do not capture groups you don't use. Read coding conventions.